File: Python/EulerTotient.py
# Github username : yatingupta10
# Website : http://www.yatingupta.me/

# Function to return gcd of a and b
def gcd(a, b):
    if a == 0:
        return b
    return gcd(b%a, a)

# A simple method to evaluate Euler Totient Function
def phi(n):
    result = 1
    for i in range(2, n):
        if gcd(i, n) == 1:
            result = result + 1
    return result

# Driver program to test above function
if __name__ == "__main__":
    for n in range(1, 11):
      print ("phi({f}) = {f1}\n".format(f = n, f1 = phi(n)))

File: Kotlin/EulerTotient.kt
fun main(args: Array<String>) {
    var result:Int = phi(25)
    println(result)                
}

private fun gcd(a: Int, b: Int): Int {
    return if (a == 0) {
        b
    } else {
        gcd(b % a, a)
    }
}

private fun phi(n: Int): Int {
    var result = 1
    for (i in 2..(n-1)) {
        if (gcd(i, n) == 1) {
            result++
        }
    }
    return result
}
